### PR TITLE
WEBAPP-498: Refactor extra to offer

### PR DESCRIPTION
**Pull request by sarahsporck**

### CODE REVIEW

File: src/endpoints/createCitiesEndpoint.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default (baseUrl: string): Endpoint<void, Array<CityModel>> => new Endpoi
     code: stripSlashes(city.path),
     live: city.live,
     eventsEnabled: city.events,
-    extrasEnabled: city.extras,
+    offersEnabled: city.extras,
```
Comment (sarahsporck) [inline src/endpoints/createCitiesEndpoint.js:27]: Only for the JsonCityModel the term extras remains, as this is set by the cms

File: src/endpoints/createOffersEndpoint.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export const OFFERS_ENDPOINT_NAME = 'extras'
+
+const createPostMap = (jsonPost: JsonOfferPostType): Map<string, string> => {
+  const map = new Map()
+  Object.keys(jsonPost).forEach(key => map.set(key, jsonPost[key]))
+  return map
+}
+
+type ParamsType = { city: ?string, language: ?string }
```
Comment (steffenkleinle) [inline src/endpoints/createOffersEndpoint.js]: I think this type is wrong, right? Should not be optional.

**Review by steffenkleinle — APPROVED** [review]

File: src/endpoints/createOffersEndpoint.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import OfferModel from '../models/OfferModel'
+import EndpointBuilder from '../EndpointBuilder'
+
+import ParamMissingError from '../errors/ParamMissingError'
+import type { JsonOfferPostType, JsonOfferType } from '../types'
+import Endpoint from '../Endpoint'
+
+export const OFFERS_ENDPOINT_NAME = 'extras'
+
+const createPostMap = (jsonPost: JsonOfferPostType): Map<string, string> => {
```
Comment (maxammann) [inline src/endpoints/createOffersEndpoint.js:11]: This can be simplified to: `new Map(Object.entries(jsonPost))`

**Review by maxammann — APPROVED** [review]

File: src/endpoints/createOffersEndpoint.js (hunk 2 above)
Comment (sarahsporck) [reply src/endpoints/createOffersEndpoint.js]: Hmm...not sure about that on the native end, but on the web app city and language are always present

File: src/endpoints/createOffersEndpoint.js (hunk 3 above)
Comment (sarahsporck) [reply src/endpoints/createOffersEndpoint.js:11]: Flow doesn't like that yet. See [here](https://github.com/facebook/flow/issues/5838).

File: src/endpoints/createOffersEndpoint.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// @flow
+
+import OfferModel from '../models/OfferModel'
+import EndpointBuilder from '../EndpointBuilder'
+
+import type { JsonOfferPostType, JsonOfferType } from '../types'
+import Endpoint from '../Endpoint'
+
+export const OFFERS_ENDPOINT_NAME = 'offers'
```
Comment (sarahsporck) [inline src/endpoints/createOffersEndpoint.js:9]: I think I need to change this name to offers (as already applied here), as it is used in the webapp. Are there any objections? @maxammann I guess this change requires a new release. How should I do this? :)

Comment (maxammann) [reply src/endpoints/createOffersEndpoint.js:9]: You are right, next version should be 5.0.1 probably. even though it is breaking.

File: src/endpoints/createOffersEndpoint.js (hunk 3 above)
Comment (maxammann) [reply src/endpoints/createOffersEndpoint.js:11]: Should we use here a @FlowFixMe?
Or is it correct that the type could be mixed?

**Pull request merged**